stored_requests/backends/file_fetcher/fetcher.go

```diff
@@ -5,126 +5,107 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
-	"strings"
+	"os"
+	"path"
+	"path/filepath"
 
+	"github.com/golang/glog"
 	"github.com/prebid/prebid-server/stored_requests"
 )
 
-// NewFileFetcher _immediately_ loads stored request data from local files.
-// These are stored in memory for low-latency reads.
-//
-// This expects each file in the directory to be named "{config_id}.json".
-// For example, when asked to fetch the request with ID == "23", it will return the data from "directory/23.json".
+// NewFileFetcher lazy-loads various kinds of objects from the given directory
+// Expected directory structure:
+// - stored_requests/{id}.json for stored requests
+// - stored_imps/{id}.json for stored imps
+// - {adserver}.json non-publisher specific primary adserver categories
+// - {adserver}/{adserver}_{account_id}.json publisher specific categories for primary adserver
 func NewFileFetcher(directory string) (stored_requests.AllFetcher, error) {
-	storedData, err := collectStoredData(directory, FileSystem{make(map[string]FileSystem), make(map[string]json.RawMessage)}, nil)
-	return &eagerFetcher{storedData, nil}, err
-}
-
-type eagerFetcher struct {
-	FileSystem FileSystem
-	Categories map[string]map[string]stored_requests.Category
+	_, err := ioutil.ReadDir(directory)
+	if err != nil {
+		return &fileFetcher{}, err
+	}
+	// read - but don't store - all the files to warm os cache
+	go filepath.Walk(directory, func(path string, info os.FileInfo, err error) error {
```

Comment:
Reading the files causes the OS to store the file data and dir metadata in its file system cache. A subsequent open/read will read from that cache so it will be faster. Each OS has its own version with similar behavior. Linux has the buffers and page cache.

That being said, it is just a nicety to avoid a slow first request with a given stored request, mainly for testing since in real traffic it matters very little, and I can remove it.

Reply:
Gotcha. I recommend to either remove it or give the host an option to warm up the cache.